ComfortZoneApp/app/src/main/java/com/example/comfortzone/Fragments/FlightFragment.java

```diff
@@ -47,21 +51,49 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
         super.onViewCreated(view, savedInstanceState);
 
         client = new WeatherClient();
+        cityList = new ArrayList<>();
+        flightsAdapter = new FlightsAdapter(getContext(), cityList);
+
+        initViews(view);
+        populateViews();
         maybeUpdateCitiesList();
     }
 
+
+
+    private void initViews(@NonNull View view) {
+        rvCities = view.findViewById(R.id.rvCities);
+    }
+
+    private void populateViews() {
+        rvCities.setAdapter(flightsAdapter);
+        rvCities.setLayoutManager(new LinearLayoutManager(getContext()));
+        client.getCityData(new CityListCallback() {
+            @Override
+            public void cityList(List<City> cities) {
+                getActivity().runOnUiThread(new Runnable() {
+                    @Override
+                    public void run() {
+                        flightsAdapter.addAll(cities);
+                    }
+                });
+
+            }
+        });
+    }
+
     public void maybeUpdateCitiesList() {
         AllWeathersDatabase db = AllWeathersDatabase.getDbInstance(getContext().getApplicationContext());
         Long timeNow = System.currentTimeMillis();
         Long hourAgo = timeNow - TimeUnit.HOURS.toMillis(1L);
         List<WeatherData> timesBeforeHour = db.weatherDao().getUploadTimes(hourAgo);
-        if (!timesBeforeHour.isEmpty()) {
+        if (db.weatherDao().getAll().isEmpty() || timesBeforeHour.isEmpty()) {
             db.weatherDao().deleteEntireTable();
-            getCities();
+            updateCities();
         }
     }
 
-    public void getCities() {
+    public void updateCities() {
```

Comment:
I think we are updating weather data rather than cities, just optional, but I think it's clearer if you refer them as weather. Additionally, since you are doing network call to get new weather data for the cities and also saving it, maybe a better method name would be `getAndStoreWeatherData`?